src/sgx/CMakeLists.txt

```diff
@@ -42,41 +41,41 @@ set(THREADS_PREFER_PTHREAD_FLAG ON)
 set(SGX_SDK_ENCLAVE_SIGNER ${SGX_SDK_PATH}/bin/x64/sgx_sign)
 set(SGX_SDK_ENCLAVE_EDGER8R ${SGX_SDK_PATH}/bin/x64/sgx_edger8r)
 
-# WAMR build configuration
-set(WAMR_BUILD_TARGET X86_64)
+# --------------------------------------------------------
+# WAMR Build
+# --------------------------------------------------------
+
+# Set target platform details
+set (WAMR_BUILD_PLATFORM "linux-sgx")
+set (WAMR_BUILD_TARGET X86_64)
 
-if(FAASM_SGX_WAMR_AOT_MODE)
+# Set WAMR include flags
+# 28/06/2021 - By default, (fast) interpreter, AOT, libc, wasi, and pthread are
+# enabled by default
+
+# Set interpreter mode
+set (WAMR_BUILD_INTERP 1)
+if (FAASM_SGX_WAMR_AOT_MODE)
+    set (WAMR_BUILD_AOT 1)
     add_definitions(-DFAASM_SGX_WAMR_AOT_MODE=1)
-    include (${WAMR_IWASM_DIR}/aot/iwasm_aot.cmake)
-else()
+else ()
+    set (WAMR_BUILD_AOT 0)
     add_definitions(-DFAASM_SGX_WAMR_AOT_MODE=0)
-    include (${WAMR_IWASM_DIR}/interpreter/iwasm_interp.cmake)
-endif()
-
-if(FAASM_SGX_WAMR_BUILDIN_LIBC)
-    include (${WAMR_IWASM_DIR}/libraries/libc-builtin/libc_builtin.cmake)
-endif()
+endif ()
+set (WAMR_BUILD_JIT 0)
 
-if(FAASM_SGX_WAMR_WASI_LIBC)
-    set(WAMR_BUILD_LIBC_WASI 1)
-    include (${WAMR_IWASM_DIR}/libraries/libc-wasi/libc_wasi.cmake)
-endif()
+# Set libraries
+if (FAASM_SGX_WAMR_BUILDIN_LIBC)
+    set (WAMR_BUILD_LIBC_BUILTIN 1)
+endif ()
+if (FAASM_SGX_WAMR_WASI_LIBC)
+    set (WAMR_BUILD_LIBC_WASI 0)
+endif ()
+set (WAMR_BUILD_LIB_PTHREAD 1)
 
-include (${WAMR_SHARED_DIR}/platform/linux-sgx/shared_platform.cmake)
-include (${WAMR_SHARED_DIR}/mem-alloc/mem_alloc.cmake)
-include (${WAMR_IWASM_DIR}/common/iwasm_common.cmake)
-include (${WAMR_SHARED_DIR}/utils/shared_utils.cmake)
-
-set(FAASM_SGX_WAMR_SRC
-        ${IWASM_AOT_SOURCE}
-        ${IWASM_COMMON_SOURCE}
-        ${IWASM_INTERP_SOURCE}
-        ${LIBC_BUILTIN_SOURCE}
-        ${LIBC_WASI_SOURCE}
-        ${MEM_ALLOC_SHARED_SOURCE}
-        ${PLATFORM_SHARED_SOURCE}
-        ${UTILS_SHARED_SOURCE}
-        )
+# Let WAMR do the including and importing of the sources
+include (${WAMR_ROOT_DIR}/build-scripts/runtime_lib.cmake)
```

Comment:
This seems to be how it's done in [WAMR](https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/product-mini/platforms/linux-sgx/CMakeLists.txt#L86)'s example.

This is, (1) set variables, (2) include just one `*.cmake` file.

Reply:
I think including that link in the source could be useful for reference in future.